src/Nest/Document/Multiple/MultiTermVectors/MultiTermVectorsRequest.cs

```diff
@@ -5,43 +5,84 @@
 
 namespace Nest
 {
+	/// <summary>
+	/// A Multi termvectors API request
+	/// </summary>
 	public partial interface IMultiTermVectorsRequest
 	{
+		/// <summary>
+		/// The documents for which to generate term vectors
+		/// </summary>
 		[JsonProperty("docs")]
 		IEnumerable<IMultiTermVectorOperation> Documents { get; set; }
+
+		/// <summary>
+		/// The ids of documents within the same index and type
+		/// for which to generate term vectors. Must be used in
+		/// conjunction with <see cref="Index"/> and <see cref="Type"/>
+		/// </summary>
+		[JsonProperty("ids")]
+		IEnumerable<Id> Ids { get; set; }
 	}
 
+	/// <inheritdoc cref="IMultiTermVectorsRequest"/>
 	public partial class MultiTermVectorsRequest
 	{
+		/// <inheritdoc />
 		public IEnumerable<IMultiTermVectorOperation> Documents { get; set; }
+
+		/// <inheritdoc />
+		public IEnumerable<Id> Ids { get; set; }
 	}
 
+	/// <inheritdoc cref="IMultiTermVectorsRequest"/>
 	[DescriptorFor("Mtermvectors")]
 	public partial class MultiTermVectorsDescriptor
 	{
-		private List<IMultiTermVectorOperation> _operations = new List<IMultiTermVectorOperation>();
+		private List<IMultiTermVectorOperation> _operations;
+
+		private List<IMultiTermVectorOperation> Operations =>
+			this._operations ?? (this._operations = new List<IMultiTermVectorOperation>());
+
 		IEnumerable<IMultiTermVectorOperation> IMultiTermVectorsRequest.Documents
 		{
-			get { return this._operations; }
-			set { this._operations = value?.ToList(); }
+			get => this._operations;
+			set => this._operations = value?.ToList();
```

Comment:
Just out of curiosity, is there any reason why we `.ToList()` on setters like this; other than to force the execution of the `IEnumerable` at this point? Seems like we could save some allocations in these instances...

Reply:
I don't know about this specific case, but in general would be to force execution as you say.

Reply:
What is unusual with respect to the general API design in NEST, is that `GetMany<T>(..)` is additive rather than assignative to the operations. I understand why this is, and it's not the only place where a method call is additive

Reply:
Places where we are additive is AFAIK only in the bulk related API helpers since you can `IndexMany`<X> and then `IndexMany<Y>` afterwards in a single bulk. That was the rationale behind the deviation. Definitely up for further discussion 😄